Improved Pokemon detail and MessageDialog

swisszeni opened the pull request:
### Fixes:

- Fixed Fonts not being loaded and applied

### Changes:

- Various UI improvements on the Pokemon detail page
- New MessageDialog
- Code refactoring

### Change details:

- Changed favorite star to look better
- Adjusted various fontfamily and fontsize 
- Resized power-up and evolve areas
- Implemented custom Message Dialog with Niantic style and animations
- New MessageDialog for Transfer, Powerup and Evolve 
- Multiple code refactoring to centralize some visual aspects in the custom.xaml

![wp_ss_20160824_0002](https://cloud.githubusercontent.com/assets/8832977/17915339/1076d82c-69aa-11e6-8ac7-9c369639a10b.png)

![wp_ss_20160824_0003](https://cloud.githubusercontent.com/assets/8832977/17915346/181fc192-69aa-11e6-8734-80f65b0fd1f9.png)

![ezgif com-video-to-gif](https://cloud.githubusercontent.com/assets/8832977/17915330/012bded0-69aa-11e6-9a01-a06c283c665e.gif)


### Other informations:

I know the MessageDialog is colliding with #1255 but I think this implementation is more flexible (variable title buttons, area for custom XAML content) and looks better (background fading and animation similar to niantics version).

Markusch85 commented:
You're right. I've closed my PR.

ST-Apps commented:
Transfer text is wrong, it shows `{0}` because you need to use `string.Format` to get the correct string. Can you please fix it?

swisszeni commented:
@ST-Apps I do use string.Format ;) The problem is, that until the last update from niantic, the transfer text was one string. Now it is a title and a text. I already updated the dialog for this change. I kept the TransferPokemonWarningText resource and added a TransferPokemonWarningTitle resource. The only thing I didn't do is updating the resources for the german language. That's why it falls back to english for the title and takes the old string (with the {0} even though there is no more placeholder in the new version) for the text. 
If you open it in english, it works perfekt (since I updated the resources there). Sorry for the confusion.

![wp_ss_20160824_0006](https://cloud.githubusercontent.com/assets/8832977/17925646/08118eea-69ee-11e6-8a98-061bf75b85e5.png)

ST-Apps commented:
Fine, merging!

EugenePavlikov commented:
I think this dialog should also be updated.

![image](https://cloud.githubusercontent.com/assets/11786389/17944064/bfb91ace-6a46-11e6-8621-c38c71537e88.png)

swisszeni commented:
Yes, but the niantic App doesn't actually use a MessageDialog for this one but something different. So when we update this dialog/code, it should be to the final design and not with 20 versions in between. Otherwise it's just a waste of time.
![screenshot_20160823-152735](https://cloud.githubusercontent.com/assets/8832977/17944346/aceec9b8-6a3f-11e6-8c66-8dc970c4c2fb.png)
